SOOUM/SOOUM/Presentations/Onboarding/NicknameSetting/OnboardingNicknameSettingViewController.swift

```diff
@@ -0,0 +1,221 @@
+    }
+    
+    private let nicknameCountLabel = UILabel().then {
+        $0.typography = .init(
+            fontContainer: BuiltInFont(
+                size: 14,
+                weight: .regular
+            ),
+            lineHeight: 19.6,
+            letterSpacing: 0
+        )
+        // TODO: - 색 수정 필요
+        $0.textColor = .som.gray500
+        $0.text = "1/8"
+    }
+    
+    private let nextButtonView = PrimaryButtonView()
+    
+    // Reactor를 연결하고, 액션과 상태를 바인딩합니다.
+    func bind(reactor: OnboardingNicknameSettingViewReactor) {
+        
+        nicknameTextField.textField.rx.text.orEmpty
+            .observe(on: MainScheduler.instance)
+            .subscribe(onNext: { [weak self] str in
+                guard let self = self else { return }
+                self.nicknameCountLabel.text = "\(str.count)/\(self.maxCount)"
+            })
+            .disposed(by: disposeBag)
+        
+        nicknameTextField.textField.rx.text.orEmpty
+            .debounce(.milliseconds(300), scheduler: MainScheduler.instance)
+            .distinctUntilChanged()
+            .map { Reactor.Action.textChanged($0) }
+            .bind(to: reactor.action)
+            .disposed(by: disposeBag)
+        
+        nicknameTextField.clearButtonView.rx.tapGesture()
+            .when(.recognized)
+            .subscribe(with: self) { object, _ in
+                object.nicknameCountLabel.text = "0/\(object.maxCount)"
+                object.nicknameTextField.textField.text?.removeAll()
+                object.reactor?.action.onNext(.textChanged(""))
+            }
+            .disposed(by: disposeBag)
+        
+        nextButtonView.rx.tapGesture()
+            .when(.recognized)
+            .subscribe(with: self) { object, _ in
+                let profileImageVC = ProfileImageSettingViewController()
+                let profileImageReactor = ProfileImageSettingViewReactor(nickname: self.nicknameTextField.textField.text!)
+                profileImageVC.reactor = profileImageReactor
+                self.navigationController?.pushViewController(profileImageVC, animated: true)
+            }
+            .disposed(by: disposeBag)
+
+        // MARK: - State Binding
+        // 닉네임 유효성 검사 결과에 따라 nextButton의 활성화 상태 업데이트
+        reactor.state
+            .map { $0.isNicknameValid ?? OnboardingNicknameSettingViewReactor.NicknameState.invalid }
+            .subscribe(with: self, onNext: { object, isValid in
+                object.nextButtonView.updateState(state: isValid == .vaild)
+                object.errorLogStackView.isHidden = isValid == .vaild
+            })
+            .disposed(by: disposeBag)
+        
+        reactor.state
+            .map { $0.errorMessage }
+            .subscribe(onNext: { [weak self] errorMessage in
+                if let message = errorMessage {
+                    self?.showErrorAlert(message)
+                }
+            })
+            .disposed(by: disposeBag)
+    }
```

Comment:
```suggestion
        
        reactor.state
            .map { $0.errorMessage }
            .subscribe(with: self) { object, errorMessage in
                if let message = errorMessage {
                    object.showErrorAlert(message)
                }
            })
            .disposed(by: disposeBag)
    }
```
위와 마찬가지로 컨벤션을 맞추기 위해 변경하면 좋을 거 같습니다.